src/python/pants/pantsd/pants_daemon.py

```diff
@@ -406,7 +406,8 @@ def run_sync(self):
     """Synchronously run pantsd."""
     # Switch log output to the daemon's log stream from here forward.
     self._close_stdio()
-    with self._pantsd_logging() as (log_stream, log_filename):
+    with self._pantsd_logging() as (log_stream, log_filename), \
+          ExceptionSink.trapped_signals(PantsDaemonSignalHandler(self)):
```

Comment:
this was failing in pantsd integration tests if I remember correctly, could you modify travis.yml to kick off pantsd integration tests for this change?